src/main/java/org/marketdesignresearch/mechlib/mechanism/auctions/Auction.java

```diff
@@ -70,191 +109,130 @@ public Good getGood(UUID id) {
+        		this.currentPhaseRoundNumber = 0;
+        	}
+        	current = this.getCurrentPhase().createNextRoundBuilder(this);
+        	this.currentPhaseRoundNumber++;
         }
-        if (bid.getBundleBids().size() > allowedNumberOfBids()) {
-            throw new IllegalBidException("Bidder " + bidder.getName() + " tried to submit more bids than allowed. Max: " + allowedNumberOfBids());
-        }
-        current.setBid(bidder, bid);
-    }
-
-    public void addRound(Bids bids) {
-        submitBids(bids);
-        closeRound();
-    }
-
+	}
+    
     public Outcome getTemporaryResult() {
-        return current.getOutcome();
+        return current.computeTemporaryResult(this.outcomeRuleGenerator);
     }
 
 
-    public Bids getAggregatedBidsAt(int round) {
+    public BB getAggregatedBidsAt(int round) {
         Preconditions.checkArgument(round >= 0 && round < rounds.size());
         return rounds.subList(0, round + 1).stream()
                 .map(AuctionRound::getBids)
-                .reduce(new Bids(), Bids::join);
+                .reduce(this::join).orElse(this.createEmptyBids());
     }
 
-    public Bids getBidsAt(int round) {
+    public BB getBidsAt(int round) {
         Preconditions.checkArgument(round >= 0 && round < rounds.size());
         return rounds.get(round).getBids();
     }
 
-    public Prices getPricesAt(int round) {
-        Preconditions.checkArgument(round >= 0 && round < rounds.size());
-        return rounds.get(round).getPrices();
-    }
-
-    public Bid getAggregatedBidsAt(Bidder bidder, int round) {
-        Preconditions.checkArgument(round >= 0 && round < rounds.size());
-        return rounds.subList(0, round + 1).stream()
-                .map(AuctionRound::getBids)
-                .map(bids -> bids.getBid(bidder))
-                .reduce(new Bid(), Bid::join);
-    }
-
-    public Bid getBidsAt(Bidder bidder, int round) {
-        return getBidsAt(round).getBid(bidder);
-    }
-
-    public Bids getLatestAggregatedBids() {
-        if (rounds.size() == 0) return new Bids();
+    public BB getLatestAggregatedBids() {
+        if (rounds.size() == 0) return this.createEmptyBids();
         return getAggregatedBidsAt(rounds.size() - 1);
     }
 
-    public Bids getLatestBids() {
-        if (rounds.size() == 0) return new Bids();
+    public BB getLatestBids() {
+        if (rounds.size() == 0) return this.createEmptyBids();
         return getBidsAt(rounds.size() - 1);
     }
-
-    public Bid getLatestAggregatedBids(Bidder bidder) {
-        if (rounds.size() == 0) return new Bid();
-        return getAggregatedBidsAt(bidder, rounds.size() - 1);
-    }
-
-    public Bid getLatestBids(Bidder bidder) {
-        if (rounds.size() == 0) return new Bid();
-        return getBidsAt(bidder, rounds.size() - 1);
-    }
-
-    public AuctionRound getRound(int index) {
+    
+    protected abstract BB join(BB b1, BB b2);
+    protected abstract BB createEmptyBids();
```

Comment:
The changed design of the bundle bid package forces to add this two abstract methods to an auction. Subtypes that now their bid type (i.e. BundleExactValueBids or BundleBoundValueBids) can then implement these methods. 
Specially the join can now avoid any typecast complete.